### PR TITLE
Add some builds tools for local minishift development

### PR DESCRIPTION
**Pull request by rhuss**
See the enclosed README for details

### CODE REVIEW

File: tools/minishift-build/rest.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+. $(dirname `realpath $0`)/vars.sh
+
+prepare_dir syndesis-rest
+mvn clean install -Pflash
+cd runtime
+mvn clean package fabric8:build -Dfabric8.mode=kubernetes -Pflash
```
Comment (zregvart) [inline tools/minishift-build/rest.sh]: isn't this the same as `-Ddeploy` ?

File: tools/minishift-build/integration-runtime.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+. $(dirname `realpath $0`)/vars.sh
+
+prepare_dir syndesis-integration-runtime
+mvn clean install
```
Comment (zregvart) [inline tools/minishift-build/integration-runtime.sh]: use `mvnw`?

File: tools/minishift-build/rest.sh (hunk 1 above)
Comment (rhuss) [reply tools/minishift-build/rest.sh]: yes, could be. Haven't checked, just put my workflow into some script form :) I will try ...

File: tools/minishift-build/integration-runtime.sh (hunk 2 above)
Comment (rhuss) [reply tools/minishift-build/integration-runtime.sh]: yeah, makes sense.

**Comment by zregvart** [comment]
I like this, but couldn't we create a repository with git subtrees and a cross platform build tool to automate it instead of shell scripts? We can even call that a poor man's _monorepo_ 😉

**Comment by rhuss** [comment]
good idea with git subtrees (we could even put it into this repo to work against repo inflation. There is still enough space ;-)

@zregvart wdym with cross platform build tool ? Aren't shell scripts not cross-platform enough ? (except for windows ;-)

**Comment by zregvart** [comment]
> @zregvart wdym with cross platform build tool ? Aren't shell scripts not cross-platform enough ? (except for windows ;-)

perhaps Gradle? would be simple to port shell scripts over, we should ideally support all platforms minishift supports, which _sic_ includes Windows

**Comment by rhuss** [comment]
we would still need platform dependent CLI tools like 'oc', 'minishift' and 'docker'. I don't see a big advantage of scripting via gradle over scripting via shell (if we can leave out windows or require sth like cygwin).

**Comment by zregvart** [comment]
very true, (but) if we start with Gradle (or something similar) we can extend this to download the required binaries (are `oc` and `docker` required for fmp, I thought it used the API directly), but not a blocker, can be discussed and (perhaps) worked on later...

**Comment by rhuss** [comment]
ok, let's merge this and see how we can expand on it maybe. Running a full 'build-all.sh' from time to  time is probably not so a bad idea ;-)

**Review by zregvart — APPROVED** [review]

**Pull request merged**